interfaces/AW-App/src/app/validation-components/validate-program/validate-program.component.ts

```diff
@@ -15,35 +17,202 @@ import { SessionStorageService } from 'src/app/services/session-storage.service'
+      this.getProgramQuestionsAndAnswers();
       this.sessionStorageService.destroyItem(this.sessionStorageService.type.scannedDid);
     });
   }
 
+  private async getProgramQuestionsAndAnswers() {
+    this.currentProgram = await this.programsService.getProgramById(this.programId);
+    await this.prepareProgramDetails(this.currentProgram);
+    this.getPrefilledAnswersProgram();
+
+  }
+
+  public async prepareProgramDetails(program: Program) {
+    this.questions = this.buildQuestions(program.customCriteria);
+  }
+
+  private buildQuestions(customCriteria: Program['customCriteria']) {
+    const questions = [];
+
+    for (const criterium of customCriteria) {
+      const question: Question = {
+        id: criterium.id,
+        code: criterium.criterium,
+        answerType: criterium.answerType,
+        label: this.mapLabelByLanguageCode(criterium.label),
+        options: this.buildOptions(criterium.options),
+      };
+      questions.push(question);
+    }
+
+    return questions;
+  }
+
+  private buildOptions(optionsRaw: any[]): QuestionOption[] {
+    if (!optionsRaw) {
+      return;
+    }
+
+    const options = [];
+
+    for (const option of optionsRaw) {
+      const questionOption: QuestionOption = {
+        id: option.id,
+        value: option.option,
+        label: this.mapLabelByLanguageCode(option.label),
+      };
+      options.push(questionOption);
+    }
+
+    return options;
+  }
+
+  private mapLabelByLanguageCode(property: any) {
+    let label = property[this.languageCode];
+
+    if (!label) {
+      label = property[this.fallbackLanguageCode];
+    }
+
+    if (!label) {
+      label = property;
+    }
+
+    return label;
+  }
+
+  private getQuestionByCode(questionCode: string): Question {
+    const result = this.questions.find((question: Question) => {
+      return question.code === questionCode;
+    });
+
+    return result;
+  }
+
+  private getAnswerOptionLabelByValue(options: QuestionOption[], answerValue: string) {
+    const option = options.find((item: QuestionOption) => {
+      return item.value === answerValue;
+    });
+
+    return option ? option.label : '';
+  }
+
+  public inputAnswers($event) {
+    const questionCode = $event.target.name;
+    this.answers[questionCode] = new Answer();
+  }
+
+  public changeAnswers($event) {
+    const questionCode = $event.target.name;
+    const answerValue = $event.target.value;
+
+    const question = this.getQuestionByCode(questionCode);
+    const answer: Answer = {
+      code: questionCode,
+      value: answerValue,
+      label: answerValue,
+    };
+
+    // Convert the answerValue to a human-readable label
+    if (question.answerType === AnswerType.Enum) {
+      answer.label = this.getAnswerOptionLabelByValue(question.options, answerValue);
+    }
+
+    this.answers[questionCode] = answer;
+
+    this.checkAllQuestionsAnswered(this.answers);
+
+  }
+
+  public initialAnswers(answersProgram) {
```

Comment:
We should amend `changeAnswers`, so that we can re-use it in here for the inital-answers-loop...

Reply:
adjusted